### PR TITLE
User guide: adding instructions on how to use and update docsy as hugo module

### PR DESCRIPTION
**Pull request by deining**
Instructions on how to make use of Docsy as hugo module, as submitted via #801.
More or less a complete rewrite of the chapters `Getting started` (now: [`Installation`](https://deploy-preview-802--docsydocs.netlify.app/docs/theme-installation/)) and `Updating Docsy` (now: [`Update or Conversion of Docsy Theme`](https://deploy-preview-802--docsydocs.netlify.app/docs/updating/)) from the user guide. Given links point to the preview of the chapters on Netlify. Enjoy reading!

### CODE REVIEW

**Comment by LisaFC** [comment]
Oh wow, this is amazing work!!! I like that we give users all the possible setup options so they can get started using their preferred workflow. I'll need to do a detailed review but will try to get to that over the next week or so.

**Comment by LisaFC** [comment]
@emckean @chalin @geriom PTAL as well - I think we may need to add a setup overview and rephrase some bits but overall this is looking good.

@deining am I reading it correctly - can users just choose between the Hugo module setup and our existing setup flows without breaking anything?

**Comment by deining** [comment]
> @deining am I reading it correctly - can users just choose between the Hugo module setup and our existing setup flows without breaking anything?

Yes, absolutely.

I just made a test to verify: I cloned the docsy example, updated the submodule and ran `hugo server` for a site preview. Then I turned both the example site and the docsy theme into a hugo module and fetched the dependencies `bootstrap` and `Font-Awesome`. Afterwards I still could run the site preview successfully. This makes me very confident, that for existing sites, we won't break anything. That also means that no one is forced to to turn his installation into a module to be able to get docsy updates. Even with modules in place, existing sites can pull in latest changes from docsy and continue using their sites as before.

That said, I wouldn't recommend (though it is certainly doable) switching back and forth between hugo module and git submodule appraoch. To be clear here: I envision that the hugo module approach becomes the de facto standard for new installations. Existing sites **can** convert their sites (which is really easy IMHO) but they **don't have to**.

**Another thought:** too many installation options might be overwhelming for first time docsy, so if agree that the module approach is the new standard, I ask myself, whether we shouldn't discard the sub section [`Cloning GitHub Repo (Traditional)`](https://deploy-preview-802--docsydocs.netlify.app/docs/theme-installation/quickstart-clone-git-repo/) completely. That said, if you decide to leave it in place, I can live with that.

**Comment by deining** [comment]
> Oh wow, this is amazing work!!!

Glad to hear you like it.

> I like that we give users all the possible setup options so they can get started using their preferred workflow.

Having a lot of options is good, but might be overwhelming for novices. I'm open for discussion here.

> I'll need to do a detailed review but will try to get to that over the next week or so.

Great!

As mentioned in #801, my PR is not ready for testing, I wish it were, I guess this would help you with your review, too.
Therefore, you could do me a favour by taking the folllowing actions:
- create a new branch `feature-hugo-module` in the docsy repo, branching from `HEAD` at `master`.
- merge https://github.com/google/docsy-example/pull/155

I will then resubmit #801 on the newly created `feature-hugo-module` branch and bring it in a testable state. All further work could then  happen on this feature branch and once we are done, you could merge this branch into master.

**Comment by LisaFC** [comment]
> **Another thought:** too many installation options might be overwhelming for first time docsy, so if we agree that the module approach is the new standard, I ask myself, whether we shouldn't discard the sub section [`Cloning GitHub Repo (Traditional)`](https://deploy-preview-802--docsydocs.netlify.app/docs/theme-installation/quickstart-clone-git-repo/) completely. That said, if you decide to leave it in place, I can live with that.

I think we can definitely reorganize your suggested docs a bit so that there's a default "happy path" set up to "get started" (so as not to confuse people) and then alternatives for those who want them.

Would like to get feedback from the rest of the community on this! (particularly what we think most users will want as the default)

**Comment by LisaFC** [comment]
> As mentioned in #801, my PR is not ready for testing, I wish it were, I guess this would help you with your review, too. Therefore, you could do me a favour by taking the folllowing actions:
> 
> * create a new branch `feature-hugo-module` in the docsy repo, branching from `HEAD` at `master`.
> * merge [Bumped docsy theme to latest version docsy-example#155](https://github.com/google/docsy-example/pull/155)
> 
> I will then resubmit #801 on the newly created `feature-hugo-module` branch and bring it in a testable state. All further work could then happen on this feature branch and once we are done, you could merge this branch into master.

Done and done!

**Comment by chalin** [comment]
If use of Hugo modules is the way to go, then this seems great! Thanks @deining! I'll take a closer look as soon as I can.

**Comment by huehnerlady** [comment]
Hi, what is the update here? We are using dependabot for git submodules but since this change our build fails with the message `Error: module "github.com/FortAwesome/Font-Awesome" not found; either add it as a Hugo Module or store it in "[...]/hugo/themes".: module does not exist`

I found the documentation about the switch [here](https://deploy-preview-802--docsydocs.netlify.app/docs/updating/convert-site-to-module/) in this Pull Request, but I feel this should be released ASAP as maybe there are other people having the same problem

**Comment by deining** [comment]
This documentation is now fully in sync with #871 and ready to be reviewed.

**Comment by chalin** [comment]
Ok, thanks, I'll have a look as soon as I can (starting tomorrow).

**Comment by LisaFC** [comment]
Similarly, I'll take a look ASAP when I have a chance (probably tomorrow)

File: userguide/.gitignore
Comment (chalin) [inline userguide/.gitignore:6]: Why this addition? The User Guide build doesn't involve the use of anything under `./themes`.

**Review by chalin — APPROVED** [review]
Thanks for your efforts @deining and @LisaFC! I only have one question/comment (see inline).

I won't have the time this week for a careful walkthrough of the new pages, but I trust that they're good. Besides, I don't want to hold up the next (and first!) official release. So I'm ok with merging this sooner than later :).

File: userguide/.gitignore
Comment (deining) [reply userguide/.gitignore]: You are right, this entry is not strictly necessary, so I removed it.

**Comment by LisaFC** [comment]
OK, I'll merge in the "Docsy Example with Hugo Modules" PR and then merge this one.

**Review by chalin — APPROVED** [review]
LGTM 🚀

**Comment by LisaFC** [comment]
Done! Shall we commence making Glorious Announcements?

**Comment by chalin** [comment]
Wonderful! Well done @deining and @LisaFC! I think that we can make announcements after the release is created?

**Pull request merged**
